### PR TITLE
Convert plugin_name to plugin_id

### PR DESCRIPTION
**Pull request by d-wysocki**
I want to merge this change because...I want switch to IDs instead of names for identifying plugins.

# Impact

* [X] New migrations
* [X] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations

# Pull Request Checklist

* [X] Privileged queries and mutations are guarded by proper permission checks
* [X] Database queries are optimized and the number of queries is constant
* [X] Database migration files are up to date
* [X] The changes are tested
* [X] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/payment/gateways/braintree/plugin.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def wrapped(self, *args, **kwargs):
 
 
 class BraintreeGatewayPlugin(BasePlugin):
+    PLUGIN_ID = "mirumee.gateway.braintree"
```
Comment (maarcingebala) [inline saleor/payment/gateways/braintree/plugin.py:35]: A gateway can be many things, maybe "mirumee.payments.braintree" (and similarly for other payment plugins)?

File: saleor/plugins/avatax/plugin.py [hunk 2]
```diff
@@ -41,6 +41,8 @@
 
 class AvataxPlugin(BasePlugin):
     PLUGIN_NAME = "Avalara"
+    PLUGIN_ID = "mirumee.avalara"
```
Comment (maarcingebala) [inline saleor/plugins/avatax/plugin.py:44]: And here maybe "mirumee.taxes.avalara".

File: saleor/plugins/vatlayer/plugin.py [hunk 3]
```diff
@@ -30,6 +30,7 @@
 
 
 class VatlayerPlugin(BasePlugin):
+    PLUGIN_ID = "mirumee.vatlayer"
```
Comment (maarcingebala) [inline saleor/plugins/vatlayer/plugin.py:33]: "mirumee.taxes.vatlayer"

File: saleor/plugins/anonymize/plugin.py [hunk 4]
```diff
@@ -12,6 +12,7 @@ class AnonymizePlugin(BasePlugin):
     """Anonymize all user data in the checkout, user profile and its orders."""
 
     PLUGIN_NAME = "Anonymize"
+    PLUGIN_ID = "mirumee.taxes.anonymize"
```
Comment (maarcingebala) [inline saleor/plugins/anonymize/plugin.py:15]: This one is not related to taxes, sorry if I suggested earlier to rename it like that. No idea how to name it, maybe `mirumee.checkout.anonymize`?

Comment (maarcingebala) [reply saleor/plugins/anonymize/plugin.py:15]: Or just `mirumee.anonymize`.

File: saleor/payment/migrations/0016_auto_20200423_0314.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def change_plugin_name_to_plugin_identifier(apps, schema_editor):
+    plugins = get_plugins()
+    payment = apps.get_model("payment", "Payment")
+
+    for payment in payment.objects.all():
+        gateway = payment.gateway
+        if gateway in plugins:
+            payment.gateway = plugins[gateway]
```
Comment (tomaszszymanski129) [inline saleor/payment/migrations/0016_auto_20200423_0314.py:20]: There is already a variable for payment.gateway in line 18

File: saleor/graphql/plugins/mutations.py [hunk 6]
```diff
@@ -46,6 +46,7 @@ class Meta:
     @classmethod
     def perform_mutation(cls, root, info, **data):
         plugin_id = data.get("id")
+        plugin_name = data.get("name")
```
Comment (fowczarek) [inline saleor/graphql/plugins/mutations.py:49]: Plugin name shouldn't be in data/input?

File: saleor/payment/migrations/0016_auto_20200423_0314.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+def get_plugins():
+    plugins = get_plugins_manager().plugins
+    return {plugin.PLUGIN_NAME: plugin.PLUGIN_ID for plugin in plugins}
+
+
+def change_plugin_name_to_plugin_identifier(apps, schema_editor):
+    plugins = get_plugins()
+    payment = apps.get_model("payment", "Payment")
+
+    for payment in payment.objects.all():
```
Comment (fowczarek) [inline saleor/payment/migrations/0016_auto_20200423_0314.py:17]: We should use `iterator` instead of  `all`.

File: saleor/account/migrations/0045_auto_20200427_0425.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        f"{plugin.PLUGIN_NAME.strip().upper()}.customer_id": f"{plugin.PLUGIN_ID.strip().upper()}.customer_id"  # noqa E501
+        for plugin in plugins
+    }
+
+
+def convert_user_payments_customer_id(apps, schema_editor):
+    users = apps.get_model("account", "User").objects.exclude(private_metadata={})
+    payments_customer_ids = create_payments_customer_ids()
+
+    for user in users:
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/account/migrations/0045_auto_20200427_0425.py:22]: @d-wysocki,  @fowczarek  is right. `private_metadata` is a JSON filed which can be tricky to filter out it. But in the case of many users - filter like this `.exclude(private_metadata={})` can be very expensive. We have to make some performance tests for that.
cc @maarcingebala

File: saleor/payment/migrations/0016_auto_20200423_0314.py (hunk 7 above)
Comment (korycins) [reply saleor/payment/migrations/0016_auto_20200423_0314.py:17]: @fowczarek is right. This will explode in case of many payment objects in DB.
cc @maarcingebala
